.storybook/main.ts

```diff
@@ -33,14 +24,24 @@ module.exports = {
     ],
     staticDirs: ["../public"],
     webpackFinal: async (config) => {
-        config.resolve.modules = [
+        config.resolve!.modules = [
             path.resolve(__dirname, ".."),
             "node_modules",
         ];
         return config;
     },
-    framework: "@storybook/react-webpack5",
-    docs: {
-        autodocs: true,
+    framework: {
+        name: "@storybook/react-webpack5",
+        options: {},
     },
+    swc: () => ({
```

Comment:
can explain what this option does?

Reply:
the swc option was needed for the storybook to work. the compiler options out of the box didn't work. got this from stack overflow